schema/crdb/dbinit.sql

```diff
@@ -4132,6 +4132,154 @@ CREATE INDEX IF NOT EXISTS lookup_region_snapshot_by_snapshot_id on omicron.publ
     snapshot_id
 );
 
+
+/*
+ *Timeseries schema tables.
+ */
+
+-- The authorization scope indicates the level at which we check a user's
+-- authorization when running a timeseries query.
+CREATE TYPE IF NOT EXISTS omicron.public.timeseries_authz_scope AS ENUM (
+    -- Timeseries data is only visible to a fleet reader.
+    'fleet',
+    -- Timeseries data is limited to the silo of the user.
+    'silo',
+    -- Timeseries data is limited to the project of the user.
+    'project',
+    -- Timeseries data is viewable to all without limitation.
+    'viewable_to_all'
+);
+
+-- Indicates whether a field of a timeseries derives from the target or metric.
+CREATE TYPE IF NOT EXISTS omicron.public.timeseries_field_source AS ENUM (
+    'target',
+    'metric'
+);
+
+-- The data type of a timeseries field.
+CREATE TYPE IF NOT EXISTS omicron.public.timeseries_field_type AS ENUM (
+    'string',
+    'i8',
+    'u8',
+    'i16',
+    'u16',
+    'i32',
+    'u32',
+    'i64',
+    'u64',
+    'ip_addr',
+    'uuid',
+    'bool'
+);
+
+-- The data type of a timeseries's datum, the actual value it measures.
+CREATE TYPE IF NOT EXISTS omicron.public.timeseries_datum_type AS ENUM (
+    'bool'
+    'i8',
+    'u8',
+    'i16',
+    'u16',
+    'i32',
+    'u32',
+    'i64',
+    'u64',
+    'f32',
+    'f64',
+    'string',
+    'bytes',
+    'cumulative_i64',
+    'cumulative_u64',
+    'cumulative_f32',
+    'cumulative_f64',
+    'histogram_i8',
+    'histogram_u8',
+    'histogram_i16',
+    'histogram_u16',
+    'histogram_i32',
+    'histogram_u32',
+    'histogram_i64',
+    'histogram_u64',
+    'histogram_f32',
+    'histogram_f64'
+);
+
+-- Units of a timeseries's datum.
+CREATE TYPE IF NOT EXISTS omicron.public.timeseries_units AS ENUM (
+    'count',
+    'bytes'
+);
+
+-- Timeseries schema describe the structure of a timeseries.
+--
+-- These are defined in TOML in `oximeter/oximeter/schema`, and loaded into the
+-- database on startup.
+CREATE TABLE IF NOT EXISTS omicron.public.timeseries_schema (
+    -- The name of the timeseries.
+    --
+    -- Names are derived from the concatenation of the target and metric names,
+    -- joined with a colon (':').
+    timeseries_name STRING(128) PRIMARY KEY,
```

Comment:
We should probably add these limits into the schema generation mechanisms. Those should check the length of the data loaded from TOML, and fail if they exceed these limits. It would be unfortunate to have a "valid" name or description, which is then truncated in the database.

Reply:
I went ahead and did this in https://github.com/oxidecomputer/omicron/pull/6009. We should review and merge that first, and then I'll update the limits here with those values once we're happy with them.